server/controller/root.py

```diff
@@ -104,6 +113,60 @@ def historical():
         'page': 'historical',
     }
 
+# Placeholder
+@get('/paste')
+@kajiki_view('paste')
+@authenticated()
+def paste():
+    if not controller.user_management_interface.is_user_logged_in(request):
+        redirect('/login_page')
+    api_key = controller.user_management_interface.get_api_key_for_user(request)
+    return {'page': 'paste',
+            'raw_logs':''}
+
+
+@post('/paste')
+@kajiki_view('paste')
+def paste():
+    if not controller.user_management_interface.is_user_logged_in(request):
+        redirect('/login_page')
+
+    api_key = controller.user_management_interface.get_api_key_for_user(request)
+
+    os_type = request.forms.get('os_type')
+    upload = request.files.get('upload')
+    #file_object = io.BytesIO(upload)
+    #file_object  = open("/Users/amandagiles/Documents/"+str(upload), 'r')
+    pasted = request.forms.get('message')
+    if upload != None:
+        message = str(upload.file.read(), 'utf-8')
+        print(message)
+    elif pasted != None:
+        message = pasted
+    try:
+        parsed_message = parsing_lib.LogParser.parse(message, os_type)
+        print(parsed_message)
+        log_entries = parsing_lib.LogParser.convert_to_html(parsed_message)
+    except:
+        log_entries = ""
+        e = sys.exc_info()[0]
+        tracebackk = str(traceback.print_exc())
+        flash = {'content': "Log format error: " +str(e), 'cls': 'error'}
+        return {'page': 'paste',
+                'api_key': api_key,
+                # If you've made it here, you have to be successfully logged in
+                'logged_in': True,
+                'raw_logs': request.forms.get('message'),
+                'log_entries': Markup(log_entries),
+                'flash': flash,
+                }
+    return {'page': 'paste',
+            'api_key': api_key,
```

Comment:
no need for `api_key` or `logged_in`, it is handled by the `@authenticate` annotation.